### PR TITLE
File Explorer drag and drop to move files

### LINKED ISSUE #86: Notebook UI: Drag and Drop to move
Related to #72 

Add drag and drop behaviour to move files into File Explorer

### PR DESCRIPTION
**Pull request by mofojed**
- Add back the drag and drop to move files functionality
- Switch FileList to use ItemList instead of SingleClickItemList; this allowed removing the FileListItem type entirely and just using FileStorageItem
- Changed `keyboardIndex` to `focusIndex` in `ItemList`, and fire an `onFocusChange` event to allow handling.
- Update selection after drop
- Expand a folder by hovering over it while dragging files
- Emit signal for all renamed files
- Converted ItemList unit tests to tsx, added some more
- Don't focus a notebook panel when opening it in preview mode (cursor was getting lost in file list)
- Fixed up MockIrisGridModel so styleguide works correctly
Fixes #86

### CODE REVIEW

**Comment by mofojed** [comment]
Adding @dsmmcken to review styling

File: packages/file-explorer/src/FileList.tsx [hunk 1]
```diff
@@ -107,6 +162,31 @@ export function getItemIcon(item: FileListItem): IconDefinition {
...
+  draggedItems: FileStorageItem[],
+  targetItem: FileStorageItem
+): { files: FileStorageItem[]; targetPath: string } {
+  if (draggedItems.length === 0 || !targetItem) {
+    throw new Error('No items to move');
+  }
+
+  const targetPath = isDirectory(targetItem)
+    ? FileUtils.makePath(targetItem.filename)
+    : FileUtils.getPath(targetItem.filename);
```
Comment (vbabich) [inline packages/file-explorer/src/FileList.tsx:178]:
```suggestion
  const targetPath = getPathFromItem(targetItem);
```

**Review by vbabich — CHANGES_REQUESTED** [review]

File: packages/file-explorer/src/FileList.tsx [hunk 2]
```diff
@@ -240,31 +492,81 @@ export const FileList = React.forwardRef(
...
+        }, DRAG_HOVER_TIMEOUT);
+        return () => clearTimeout(timeout);
+      }
+    }, [dropTargetItem, table]);
+
     useImperativeHandle(ref, () => ({
       updateDimensions: () => {
         requestAnimationFrame(() => {
-          itemList.current?.updateViewport();
+          // TODO: still needed??
```
[the remark this answers is not in the document]
Comment (mofojed) [reply packages/file-explorer/src/FileList.tsx:514]: Yup I didn't need this with the new AutoSizer, I'll take out this updateDimensions wiring

**Review by dsmmcken — DISMISSED** [review]
- [x] 1. Dragging folder into self deletes folder 😨 
- [x] 2, Styling of selection should match command history exactly see commandhistoryitem.scss, maybe even re-use? note the use of focus, active, hover, focus-within active styling and borders.
- [x] 3. Selecting multiple items, then initiating a drag shows the drag item that initiated as de-selected, rather than remaining selected
- [x] 4.  ctrl/shift selection clicks shouldn't trigger opening of that notebook.

**Comment by vbabich** [comment]
- [ ] In Chrome/Mac the cursor has a green "+" icon when dragging items

**Comment by mofojed** [comment]
@dsmmcken Should I be building the border/transitions directly into ItemList? Not sure if there was a reason that styling was only added to command history or not.

In any case I've pushed a change with the styling applied to FileList

**Comment by dsmmcken** [comment]
> @dsmmcken Should I be building the border/transitions directly into ItemList? Not sure if there was a reason that styling was only added to command history or not.
> 
> In any case I've pushed a change with the styling applied to FileList

Could re-use between filelist and command history list, but this list is also used a bunch in the table sidebar menu, where it has different styling

**Review by vbabich — DISMISSED** [review]

**Review by vbabich — APPROVED** [review]

**Pull request merged**